aikau/src/main/resources/alfresco/services/SiteService.js

```diff
@@ -593,52 +627,275 @@ define(["dojo/_base/declare",
+
+      /**
+       * This method delegates edit creation to the legacy YUI popup as found in Alfresco Share unless the 
+       * [legacyMode]{@link module:alfresco/services/SiteService#legacyMode} is configured to be false, in which
+       * case a request is published to get the full site details in order to display a form dialog containing the 
+       * [widgetsForEditSiteDialog]{@link module:alfresco/services/SiteService#widgetsForEditSiteDialog}.
        *
        * @instance
        * @param {string} site
        */
       editSite: function alfresco_services_SiteService__editSite(config) {
-         // TODO: When an edit site request is received we should display the edit site dialog.
-         //       We need to wrap the existing YUI widget in a Dojo object.
          this.alfLog("log", "A request has been made to edit a site: ", config);
+         if (this.legacyMode)
+         {
+            /* global Alfresco */
+            if (Alfresco && Alfresco.module && typeof Alfresco.module.getEditSiteInstance === "function")
+            {
+               Alfresco.module.getEditSiteInstance().show({
+                  shortName: config.site
+               });
+            }
+            else
+            {
+               this.alfLog("error", "Could not find the 'Alfresco.module.getEditSiteInstance' function - has 'edit-site.js' been included in the page?");
+            }
+         }
+         else
+         {
+            // Get the site data to display...
+            var url = AlfConstants.PROXY_URI + "api/sites/" + config.site;
+            this.serviceXhr({url : url,
+                             method: "GET",
+                             site: config.site,
+                             successCallback: this.showEditSiteDialog,
+                             callbackScope: this});
+         }
+      },
 
-         // Just use the old YUI popup...
-         if (Alfresco && Alfresco.module && typeof Alfresco.module.getEditSiteInstance === "function")
+      /**
+       * This is called as a result of [editSite]{@link module:alfresco/services/SiteService#editSite} being used
+       * when [legacyMode]{@link module:alfresco/services/SiteService#legacyMode} is configured to be false. It 
+       * takes the site data supplied in the response and publishes a request to display a form dialog containing the 
+       * [widgetsForEditSiteDialog]{@link module:alfresco/services/SiteService#widgetsForEditSiteDialog}.
+       *
+       * @instance
+       * @param {object} response The details of the site to be edited
+       * @param {object} originalRequestConfig The details of the request made to obtain site data
+       * @fires module:alfresco/core/topics#CREATE_FORM_DIALOG
+       */
+      showEditSiteDialog: function alfresco_services_SiteService__showEditSiteDialog(response, originalRequestConfig) {
+         // Check that the resposne is the expected siteData...
+         var shortName = lang.getObject("shortName", false, response);
```

Comment:
No, I was using this to just check that the response maps to the site data as best as I could.